widget/entry.go

```diff
@@ -461,12 +477,66 @@ func (e *Entry) Tapped(ev *fyne.PointEvent) {
 	e.Lock()
 	e.CursorRow = row
 	e.CursorColumn = col
+	if startSelect {
+		e.selectRow = row
+		e.selectColumn = col
+	}
 	e.Unlock()
 	Renderer(e).(*entryRenderer).moveCursor()
 }
 
-// TappedSecondary is called when right or alternative tap is invoked - this is currently ignored.
-func (e *Entry) TappedSecondary(_ *fyne.PointEvent) {
+// DoubleTapped is called when this entry has been double tapped so we should select text below the pointer
+func (e *Entry) DoubleTapped(ev *fyne.PointEvent) {
+	row := e.textProvider().row(e.CursorRow)
+	start, end := e.CursorColumn, e.CursorColumn+1
+	if start >= len(row) {
+		start = len(row) - 1
+	}
+	if end >= len(row) {
+		end = len(row) - 1
+	}
+
+	if row[start] == ' ' || row[start] == '\t' {
```

Comment:
This line will crash if you double tap an empty entry

Reply:
Yeah, I tested for this and experienced crashes. Lines 492-497 prevent start and end from causing an out-of-bounds test on the row.

Reply:
Yes, when len(row) is 0 the guards fail